core/invoker/src/main/scala/whisk/core/containerpool/kubernetes/KubernetesClient.scala

```diff
@@ -0,0 +1,222 @@
+      }
+    }.recoverWith {
+      case e =>
+        log.error(this, s"Failed to get IP of Pod '${id.asString}' within timeout: ${e.getClass} - ${e.getMessage}")
+        Future.failed(new Exception(s"Failed to get IP of Pod '${id.asString}'"))
+    }
+  }
+
+  def rm(id: ContainerId)(implicit transid: TransactionId): Future[Unit] =
+    runCmd(Seq("delete", "--now", "pod", id.asString), timeouts.rm).map(_ => ())
+
+  def rm(key: String, value: String)(implicit transid: TransactionId): Future[Unit] =
+    runCmd(Seq("delete", "--now", "pod", "-l", s"$key=$value"), timeouts.rm).map(_ => ())
+
+  def logs(id: ContainerId, sinceTime: String)(implicit transid: TransactionId): Source[ByteString, Any] = {
+
+    Source.fromFuture(Future {
+      blocking {
+        val sinceTimePart = if (!sinceTime.isEmpty) {
+          s"&sinceTime=${sinceTime}"
+        } else ""
+        val url = new URL(
+          URLUtils.join(
+            kubeRestClient.getMasterUrl.toString,
+            "api",
+            "v1",
+            "namespaces",
+            kubeRestClient.getNamespace,
+            "pods",
+            id.asString,
+            "log?timestamps=true" ++ sinceTimePart))
+        val request = new Request.Builder().get().url(url).build
+
+        val response = kubeRestClient.getHttpClient.newCall(request).execute
+        if (!response.isSuccessful) {
+          Future.failed(
+            new Exception(s"Kubernetes API returned HTTP status ${response.code} when trying to retrieve pod logs"))
+        }
+        response.body.string
+      }
+    }.map { output =>
+      val original = output.lines.toSeq
+      val relevant = original.dropWhile(s => !s.startsWith(sinceTime))
+      val result =
+        if (!relevant.isEmpty && original.size > relevant.size) {
+          // drop matching timestamp from previous activation
+          relevant.drop(1)
+        } else {
+          original
+        }
+      // map the logs to the docker json file format expected by
+      // ActionLogDriver.processJsonDriverLogContents
+      // TODO - jcrossley - you'll probably want to clean up / reimplement the logic intended by this with the new format
+      //var sentinelSeen = false
+      ByteString(
+        result
+          .map { line =>
+            val pos = line.indexOf(" ")
+            val ts = line.substring(0, pos)
+            val msg = line.substring(pos + 1)
+            // TODO - can we get the proper stream name from kubernetes? Some stuff is stderr
+            val stream = "stdout"
+            JsObject("log" -> msg.toJson, "stream" -> stream.toJson, "time" -> ts.toJson)
+          }
+          .mkString("\n") + "\n" // trailing newline is necessary or the frame won't be decoded and break the akka stream
+      )
+    })
+  }
+
+  private def runCmd(args: Seq[String], timeout: Duration)(implicit transid: TransactionId): Future[String] = {
+    val cmd = kubectlCmd ++ args
+    val start = transid.started(
```

Comment:
Recent changes to this logger method means this is only emitted at debug level. FYI.

Reply:
Looks like `DockerContainer` explicitly sets `InfoLevel` now. I'll update to match.